Let's try InstantOn on Intel Mac

una-tapa opened the pull request:
This article explains how to try InstantOn on IntelMac environment. 

The English version for the Liberty blog admins. 
https://github.com/OpenLiberty/blogs/blob/f12546fb14d1c9560efeae22b61fca9b7b46bf07/posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc

The Japanese version for the local team
https://github.com/OpenLiberty/blogs/blob/f12546fb14d1c9560efeae22b61fca9b7b46bf07/posts/ja/2022-11-01-letstry-instant-on-intel-mac-ja.adoc

yeekangc commented:
I like the idea of such a blog post.  It helps developers to try out InstantOn on their development environments (on Mac).

Cc @dmuelle @lauracowen @GraceJansen @tjwatson

yeekangc commented:
I would like to ask Tom to review the (English) content from a technical standpoint (and then, we can adjust the Japanese source if appropriate).

David can help and provide ID feedback (on the English version)?

yeekangc commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+---
+layout: post
+title: Let's try Open Liberty InstantOn on Intel Mac
```
```suggestion
title: Let's try Liberty InstantOn on Intel Mac
```

yeekangc commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+author_github: https://github.com/takakiyo
+seo-title: Let's try Open Liberty InstantOn on Intel Mac - OpenLiberty.io
+seo-description: Let's try Open Liberty InstantOn on Intel Mac
+blog_description: "Let's try Open Liberty InstantOn on Intel Mac"
```
```suggestion
blog_description: "Let's try Liberty InstantOn on Intel Mac"
```

yeekangc commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/takakiyo
+author_github: https://github.com/takakiyo
+seo-title: Let's try Open Liberty InstantOn on Intel Mac - OpenLiberty.io
```
```suggestion
seo-title: Let's try Liberty InstantOn on Intel Mac - OpenLiberty.io
```

yeekangc commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+author_picture: https://avatars3.githubusercontent.com/takakiyo
+author_github: https://github.com/takakiyo
+seo-title: Let's try Open Liberty InstantOn on Intel Mac - OpenLiberty.io
+seo-description: Let's try Open Liberty InstantOn on Intel Mac
```
```suggestion
seo-description: Try Liberty InstantOn on a developer machine like a Mac system and see how you can achieve fast startup for cloud-native Java applications without compromise.
```

yeekangc commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+
+== What is Liberty InstantOn ? 
+
+Liberty InstantOn was announced on September 29, 2022. It is a technology that launches a full-featured Java EE application server in hundreds of milliseconds.
```
```suggestion
Liberty InstantOn was announced on September 29, 2022. It is a technology that launches a full-featured cloud-native Java runtime supporting MicroProfile and Jakarta EE APIs in hundreds of milliseconds.
```

tjwatson replied:
YK's rewording is better.  But the current plan for InstantOn support is to only support Jakarta WebProfile and MicroProfile.

Something is missing here with respect to InstantOn for the application.  We are not trying to position InstantOn as a fast way to start the runtime.  Rather it should be focused on fast startup of the cloud-native application itself.  I suggest:

"It is a technology that provides a full-featured cloud-native Java runtime supporting MicroProfile and Jakarta WebProfile APIs that enables applications to start in hundreds of milliseconds."

una-tapa replied:
Line 26 will be updated with Tom's suggestion
"It is a technology that provides a full-featured cloud-native Java runtime supporting MicroProfile and Jakarta WebProfile APIs that enables applications to start in hundreds of milliseconds."

yeekangc commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+
+When using Serverless technology in a container environment, the startup time is important. The response time includes the time it takes for the container process to start.  So the startup time directly affects the quality of service.
+
+Many efforts have been made to reduce the startup time for Java applications. For example, one of the efforts is to use the capabilities of GraalVM to transform the application into a native executable. However, GraalVM does not allow all Java features to be used, so it is not possible to convert widely used Java EE applications into executables.
```
```suggestion
Many efforts have been made to reduce the startup time for Java applications. For example, one of the efforts is to use the capabilities of GraalVM to transform the application into a native executable. However, GraalVM does not allow all Java features to be used, so it is not possible to convert widely used Java EE or Jakarta EE applications into executables.
```

yeekangc commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+
+Liberty InstantOn takes a totally different approach to reduce the startup time. It takes advantage of the Linux CRIU (Checkpoint/Restore In Userspace) technology. CRIU is a set of functions for recording (Checkpoint) the information of the running process in a file and restoring (Restore) the process from the file. 
+
+Open Liberty/WebSphere Liberty with an OpenJ9 VM incorporates CRUI technology to get a Checkpoint when the application finishes initializing. By committing the container in this state and resuming it from there, it can be started in a lightning fast time.
```
```suggestion
Open Liberty (and WebSphere Liberty) with an Eclipse OpenJ9 VM incorporates CRIU technology to get a Checkpoint when the application finishes initializing. By committing the container in this state and resuming it from there, it can be started in a lightning fast time.
```

tjwatson replied:
I would not mention WebSphere Liberty for an Open Liberty blog.  Also, we don't have any InstantOn support for WebSphere Liberty yet.  InstantOn is only in beta at this time and WebSphere Liberty does not have betas.

vijaysun-omr replied:
I don't believe `C` needs to be capitalized in "Checkpoint"

una-tapa replied:
Will remove WebSphere Liberty,  and change Checkpoint to checkpoint.

yeekangc commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+
+The InstantOn steps in the blog article link:https://openliberty.io/blog/2022/09/29/instant-on-beta.html[Liberty InstantOn startup for cloud native Java applications] are  based on the Red Hat Enterprise Linux. The same experience can be achieved on an Intel Mac with macOS Monterey (version 12.x). This article shares the steps to experience the exciting technology on Mac. 
+
+The InstantOn sample environment is published as a container image from IBM Cloud Registry. As the container execution environment, we can use podman introduced by Homebrew. On Mac, we needed to tweak the steps a little bit to see the InstantOn in action. 
```
```suggestion
The InstantOn sample environment is published as a container image from IBM Cloud Registry. As the container execution environment, we can use podman, which one can install via Homebrew. On Mac, we needed to tweak the steps a little bit to see the InstantOn in action. 
```

dmuelle commented:
I'm happy to review for ID after the technical reviewers have had a chance to provide feedback

tjwatson commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+
+== Let's try InstantOn on Intel Mac
+
+The InstantOn steps in the blog article link:https://openliberty.io/blog/2022/09/29/instant-on-beta.html[Liberty InstantOn startup for cloud native Java applications] are  based on the Red Hat Enterprise Linux. The same experience can be achieved on an Intel Mac with macOS Monterey (version 12.x). This article shares the steps to experience the exciting technology on Mac. 
```
This begs the question.  Can this work with M1 Rosetta? If we know it doesn't work then we should mention it.

una-tapa replied:
Currently trying to chat with the techsales team. leaving the conversation open.  Will go through the doc and remove all "WAS Liberty" mention.

una-tapa replied:
Adding "Currently, InstantOn feature is not available on M1 aarch64 architecture. "

tjwatson commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+$ podman system connection default rootful-root
+----
+
+Following the version of the podman used for this article. 
```
```suggestion
The Following version of podman used for this article. 
```

una-tapa replied:
Thanks for catching.  Just realized it is also missing "was".  
The Following version of podman **was** used for this article.  
After committing the suggestion I will remember to add it.

tjwatson commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+[AUDIT   ] CWWKE0036I: The server defaultServer stopped after 2 minutes, 32.806 seconds.  
+----
+
+==== Getting the Checkpoint
```
```suggestion
==== Checkpoint the application
```

tjwatson commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+/opt/ol/wlp/bin/server: line 946:   130 Killed                  "${JAVA_CMD}" "$@" >> "${CHECKPOINT_CONSOLE_LOG}" 2>&1 < /dev/null 
+----
+
+Since the container was started without `--rm`` this time, the container environment after execution remains as it is. In this environment, files that record process information with checkpoint are saved.
```
```suggestion
This starts the application container and proceeds to start the application. After the application has started the runtime will perform a checkpoint of the process. When the process state has been saved the container exits. Since the container was started without `--rm` this time, the stopped container remains available for inspection. This stopped container, named `getting-started-checkpoint-container`, contains the checkpoint process state information.
```

tjwatson commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+940fd476eccc  localhost/getting-started:latest  /opt/ol/wlp/bin/s...  4 minutes ago  Exited (0) 4 minutes ago              getting-started-checkpoint-container
+----
+
+The following command commits and register it as a new image.
```
```suggestion
The following command commits the container into a new image.  This new image, named `getting-started-instanton`, will contain the checkpoint process state for the application. This process state is then used to quickly start the application when the `getting-started-instanton` image is run as a container.
```

tjwatson commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+
+== Starting Open Liberty using the InstantOn feature 
+
+Now it's time to execute InstantOn. When you start Liberty using this image, the server will start in a flash.
```
```suggestion
Now it's time to see how fast the `getting-started-instanton` application can start. When you start the application container using this image, the application will start in a flash.
```

tjwatson commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,280 @@
+
+[source]
+----
+% podman run -it --rm --privileged -p 9080:9080 getting-started-instanton
```
Have you tried this without `--privileged`.  What about:
```
podman run \
  --rm \
  --cap-add=CHECKPOINT_RESTORE \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_PTRACE \
  -p 9080:9080 \
  getting-started-instanton
```

vijaysun-omr replied:
Yes I had the same question as Tom, it would be good to have the "unprivileged restore" being mentioned here in some way (either as something you tried and it worked, OR as something that is possible on Linux (as discussed in the original Liberty InstantOn blog) but needs further investigation on Mac).

una-tapa replied:
Currently trying to chat with the techsales team. leaving the conversation open.

una-tapa replied:
I was able to try and restore worked. The content was updated with more restore without --privileged info. Spell corrected

tjwatson reviewed (commented):
I think we need some words of caution added to this blog.  Something like:

Building and running an application container image with InstantOn on a Mac is possible, but we have not done extensive testing to prove such an application container image is portable for deploying in various cloud environments.  For example, it may not be possible to deploy such a container in Red Hat OpenShift Container Platform. This article is intended to provide an easy guide to evaluating the technology locally on the Mac.  Some issues may be encountered when trying to run the produced InstantOn application container on something other than a Mac.

tjwatson commented:
I wonder how much different an article would be for Windows 10.

una-tapa commented:
@yeekangc @tjwatson @dmuelle 

Thanks a lot for the review!   
I will pass the comments to the local team for their doc, and I plan to update the Japanese draft for the OL blog as well. 
FYI - The techsales team is also trying to get InstantOn steps for demo on WSL (Windows Subsystem Linux) .

una-tapa commented:
TODO for me 
Line 26
Liberty InstantOn was announced on September 29, 2022. It is a technology that launches a full-featured cloud-native Java runtime supporting MicroProfile and Jakarta EE APIs in hundreds of milliseconds.

"It is a technology that provides a full-featured cloud-native Java runtime supporting MicroProfile and Jakarta WebProfile APIs that enables applications to start in hundreds of milliseconds."

Line 30 - Checkpoint to checkpoint 
Line 34 -Remove all "WAS Liberty"  Checkpoint checkpoint
Line 70 - Add "was"

Add a mention abut M1 aarch64
Currently, InstantOn feature is not available on M1 aarch64 architecture.

una-tapa commented:
All comments have been incorporated in the latest commit. (English only) 

Line 3,8,9,10,30,40 - YK 
26 - YK + Tom
34 - YK, Tom (Remove WAS Liberty) + Vijay (Checkpoint -> checkpoint) 
38 - Currently, InstantOn feature is not available on M1 aarch64 architecture.
70 - Tom  
Added checkpoint restore section
Updated podman version as I confirmed all the operations on my Mac
The restore section may be shortened? Suggestions appreciated. 

Diff with the previous commit: 
https://github.com/OpenLiberty/blogs/pull/2670/commits/bea6b0a3b501e1b50ff49b9e10dc62acc202d38a

Once English version is final. I will update the Japanese version accordingly.

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+
+== What is Liberty InstantOn ? 
+
+Liberty InstantOn was announced on September 29, 2022. It is a technology that provides a full-featured cloud-native Java runtime supporting MicroProfile and Jakarta WebProfile APIs that enables applications to start in hundreds of milliseconds.
```
```suggestion
Liberty InstantOn provides a full-featured cloud-native Java runtime that supports the MicroProfile and Jakarta WebProfile APIs and enables applications to start in milliseconds.
```

dmuelle replied:
"hundreds of milliseconds" seems odd but might be fine. I first read it as "hundredths of milliseconds" but I dont know what the actual figure is- what if we just say "milliseconds" ?

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+
+Liberty InstantOn was announced on September 29, 2022. It is a technology that provides a full-featured cloud-native Java runtime supporting MicroProfile and Jakarta WebProfile APIs that enables applications to start in hundreds of milliseconds.
+
+When using Serverless technology in a container environment, the startup time is important. The response time includes the time it takes for the container process to start.  So the startup time directly affects the quality of service.
```
maybe 
```suggestion
When using serverless technology in a container environment, the startup time is important. Faster startup times mean faster response times. So the startup time directly affects the quality of service.
```

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+
+Liberty InstantOn takes a totally different approach to reduce the startup time. It takes advantage of the Linux CRIU (Checkpoint/Restore In Userspace) technology. CRIU is a set of functions for recording (Checkpoint) the information of the running process in a file and restoring (Restore) the process from the file. 
+
+Open Liberty with an Eclipse OpenJ9 VM incorporates CRIU technology to get a checkpoint when the application finishes initializing. By committing the container in this state and resuming it from there, it can be started in a lightning fast time.
```
```suggestion
Open Liberty with an Eclipse OpenJ9 VM incorporates CRIU technology to get a checkpoint when the application finishes initializing. By committing the container in this state and resuming it from there, it can start in lightning fast time.
```

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+
+== Let's try InstantOn on Intel Mac
+
+The InstantOn steps in the blog article link:https://openliberty.io/blog/2022/09/29/instant-on-beta.html[Liberty InstantOn startup for cloud native Java applications] are based on the Red Hat Enterprise Linux. The same experience can be achieved on an Intel Mac with macOS Monterey (version 12.x). This article shares the steps to experience the exciting technology on Mac. Currently, InstantOn feat ...
```
```suggestion
The InstantOn steps in the  link:https://openliberty.io/blog/2022/09/29/instant-on-beta.html[Liberty InstantOn startup for cloud native Java applications] blog post are based on Red Hat Enterprise Linux. You can achieve the same experience on an Intel Mac with macOS Monterey (version 12.x). This post shares the steps to experience the exciting technology on Mac. Currently, the InstantOn feature is not available on M1 aarch64 architecture. 
```

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+
+The InstantOn steps in the blog article link:https://openliberty.io/blog/2022/09/29/instant-on-beta.html[Liberty InstantOn startup for cloud native Java applications] are based on the Red Hat Enterprise Linux. The same experience can be achieved on an Intel Mac with macOS Monterey (version 12.x). This article shares the steps to experience the exciting technology on Mac. Currently, InstantOn feat ...
+
+The InstantOn sample environment is published as a container image from IBM Cloud Registry. As the container execution environment, we can use podman, which one can install via Homebrew by `brew install podman`. On Mac, we needed to tweak the steps a little bit to see the InstantOn in action.
```
```suggestion
The InstantOn sample environment is published as a container image from IBM Cloud Registry. We can use Podman as the container execution environment, which you can install via Homebrew by  running the`brew install podman` command. On Mac, we needed to tweak the steps a little bit to see the InstantOn in action.
```

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+
+=== Preparing the Podman
+
+Since InstantOn uses the CAP_CHECKPOINT_RESTORE function of the Linux kernel, the InstantOn feature does not work properly if podman is started in rootless mode, Also, once the virtual environment (podman machine) has been started in rootless mode, it will fail even if it is run in rootful mode. The solution is to create a new dedicated virtual machine. It may be necessary to remove existing VM b ...
```
```suggestion
Since InstantOn uses the CAP_CHECKPOINT_RESTORE function of the Linux kernel, the InstantOn feature does not work properly if Podman is started in rootless mode. Also, once the virtual environment (Podman machine) is started in rootless mode, it will fail even if it is run in rootful mode. The solution is to create a new dedicated virtual machine. It might be necessary to remove any existing VMs by running the  `podman machine list` and `podman machine rm` commands. 
```

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+
+Since InstantOn uses the CAP_CHECKPOINT_RESTORE function of the Linux kernel, the InstantOn feature does not work properly if podman is started in rootless mode, Also, once the virtual environment (podman machine) has been started in rootless mode, it will fail even if it is run in rootful mode. The solution is to create a new dedicated virtual machine. It may be necessary to remove existing VM b ...
+
+The followng steps create a podman virtual environment and start it in rootful mode. We will configure it to connect to a virtual machine by default.
```
```suggestion
The followng steps create a Podman virtual environment and start it in rootful mode. We will configure it to connect to a virtual machine by default.
```

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+
+=== Create a container image
+
+To use the InstantOn image, rewrite the first line of the Dockerfile in the same directory with `beta-instanton`` tag. If you want to play around, look inside this image and do your experiment! 
```
since the examples use podman, should it be"Containerfile" instead of "Dockerfile" ? I'm not sure

una-tapa replied:
Dockerfile is what the OL Guide sample provides. Podman can handle Dockerfile. 

https://docs.podman.io/en/latest/markdown/podman-build.1.html
> podman build Builds an image using instructions from one or more Containerfiles or Dockerfiles and a specified build context directory.

una-tapa replied:
Actually I should point to the original InstantOn-beta blog. It use the same OL guide sample that uses Dockerfile.
https://openliberty.io/blog/2022/09/29/instant-on-beta.html#dockerfile

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+/opt/ol/wlp/bin/server: line 946:   130 Killed                  "${JAVA_CMD}" "$@" >> "${CHECKPOINT_CONSOLE_LOG}" 2>&1 < /dev/null 
+----
+
+This starts the application container and proceeds to start the application. After the application has started the runtime will perform a checkpoint of the process. When the process state has been saved the container exits. Since the container was started without `--rm` this time, the stopped container remains available for inspection. This stopped container, named `getting-started-checkpoint-con ...
```
```suggestion
This starts the application container and proceeds to start the application. After the application starts, the runtime performs a checkpoint of the process. When the process state has been saved, the container exits. Since the container was started without `--rm` this time, the stopped container remains available for inspection. This stopped container, named `getting-started-checkpoint-container`, contains the checkpoint process state information.
```

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+
+Running fully privileged containers is not recommended. The best practice is to instead reduce the elevated privileges down to only what is required to run the container. 
+
+It is possible to run podman with unconfined --security-opt options below also worked on the Mac OS mentioned below. For more details of the parameters and the input file, refer to link:https://openliberty.io/blog/2022/09/29/instant-on-beta.html[the original InstantOn blog] .
```
```suggestion
You can run `podman` with the following unconfined --security-opt options, which also worked on the Mac OS as mentioned after the example. For more details of the parameters and the input file, refer to link:https://openliberty.io/blog/2022/09/29/instant-on-beta.html[the original InstantOn blog].
```

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+
+It is possible to run podman with unconfined --security-opt options below also worked on the Mac OS mentioned below. For more details of the parameters and the input file, refer to link:https://openliberty.io/blog/2022/09/29/instant-on-beta.html[the original InstantOn blog] .
+
+podman command with unconfined `--security-opt` options
```
```suggestion
The following example shows the `podman` command with unconfined `--security-opt` options
```

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
````diff
@@ -0,0 +1,342 @@
+[AUDIT   ] CWWKF0011I: The defaultServer server is ready to run a smarter planet. The defaultServer server started in 0.283 seconds
+```
+
+podman command with seccomp to only required system calls. 
````
```suggestion
The following example shows the `podman`  command with `seccomp` to only required system calls. 
```

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
````diff
@@ -0,0 +1,342 @@
+[AUDIT   ] CWWKF0011I: The defaultServer server is ready to run a smarter planet. The defaultServer server started in 0.262 seconds.
+```
+
+The podman command may no longer need to mount `ns_last_pid` nor `seccomp` parameter at a newer OS version. The following CHECKPOINT_RESTORE command worked on macOS Monterey 12.6.  
````
```suggestion
The podman command might not  need to mount `ns_last_pid` nor `seccomp` parameter at a newer OS version. The following CHECKPOINT_RESTORE command worked on macOS Monterey 12.6.  
```

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
````diff
@@ -0,0 +1,342 @@
+  -p 9080:9080 \
+  getting-started-instanton
+```
+The above command ran successfully on the following MacOS version. 
````
```suggestion
The previous command ran successfully on the following MacOS version. 
```

dmuelle reviewed (commented):
ID review- left a few suggestions. One general question I have is whether we want to make it more clear at the outset that InstantOn is currently a beta feature

una-tapa commented:
@dmuelle 

Thanks for all the comments to make the article natural for the native readers.   
I committed all of them but one about "Dockerfile" which is also used in the original beta blog. 

>  One general question I have is whether we want to make it more clear at the outset that InstantOn is currently a beta feature

I agree. Could you suggest where and what phrase to add?

vijaysun-omr commented:
"Liberty InstantOn was announced as a beta feature on September 29, 2022." is one possible phrasing

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+
+== What is Liberty InstantOn ? 
+
+Liberty InstantOn provides a full-featured cloud-native Java runtime that supports the MicroProfile and Jakarta WebProfile APIs and enables applications to start in milliseconds.
```
```suggestion
Liberty InstantOn provides a full-featured cloud-native Java runtime that supports the MicroProfile and Jakarta WebProfile APIs and enables applications to start in milliseconds. InstantOn is currently available as a special beta release container image.
```

una-tapa replied:
This is also great.  The commit suggestion button is grayed out at this time. I will come back again.

una-tapa replied:
My update including Vijay's comment crossed with David's suggestion. I pushed his suggested change.  Is "beta" twice a bit repetitive? I will wait for review.

dmuelle commented:
I made a suggestion.

> InstantOn is currently available as a special beta release container image.

 I don't know that the specific date will be relevant to readers, other than to show the release is recent. I guess the main concern is to show that InstantOn is still in beta and not available in regular Liberty releases yet. One thing to consider is that while InstantOn is a beta, it is separate from the regular Open Liberty beta release, which is replaced each month with a new beta. I'm not sure how best to capture that, or if we even need to.
 
 I'm not sure I have the wording quite right 
 @yeekangc @mbroz2 what do you think?

una-tapa commented:
Updated the article with Vijay's suggestion, thanks.

una-tapa commented:
Just so, I got very curious if InstantOn feature runs on WSL2 (Subsystem Linux Ubuntu 20.04) on Windows 11.  
After several hacks pointed from the RedHat article below, I made progress in installing podman but I see more hoops to jump through. I decided it is not worth trying further at this time. Not sure if it has any value add to mention in the article. Maybe not. 

--- Details ---
By following the redhat doc below. I was able to install podman. 
- https://www.redhat.com/sysadmin/run-podman-windows

To install, several "hacks" were required.  
- https://dev.to/thangchung/start-podman-on-wsl2-in-4-steps-3jn9
- https://phoenixnap.com/kb/install-podman-on-ubuntu

The podman version installed on WSL2 is 3.4.2 that is  old compared with podman on my Intel Mac (4.3.0) 
I ran into [an issue](https://github.com/containers/podman/issues/12161) when I started podman `podman machine start`. 
The suggested workaround did not work for me. 

```
$ podman version
Version:      3.4.2
API Version:  3.4.2
Go Version:   go1.15.2
Built:        Wed Dec 31 19:00:00 1969
OS/Arch:      linux/amd64
```

vijaysun-omr commented:
I agree that the mention of Windows 11 experimentation with the Linux sub system is probably out of scope for this article. I believe Ian had gotten Windows/Podman configuration to work but I don't know details.

una-tapa commented:
Both Vijay and David's comments are included in the draft to get input for the reviewers.

kinueng commented:
Hi all,
Are we ready to publish this blog post?

dmuelle commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+
+== What is Liberty InstantOn ? 
+
+Liberty InstantOn was announced as a beta feature on September 29, 2022. It is a technology that provides a full-featured cloud-native Java runtime supporting MicroProfile and Jakarta WebProfile APIs that enables applications to start in hundreds of milliseconds. InstantOn is currently available as a special beta release container image.
```
```suggestion
Liberty InstantOn provides a full-featured cloud-native Java runtime that enables applications to start in hundreds of milliseconds and supports MicroProfile and Jakarta WebProfile APIs. InstantOn is currently available as a special beta release container image.
```

una-tapa commented:
Hi, @kinueng 
Thanks for catching this blog post!  
The Japanese version has been updated to include all the comments received on the English version.

yeekangc commented on posts/ja/2022-11-01-letstry-instant-on-intel-mac-en.adoc:
```diff
@@ -0,0 +1,342 @@
+author_picture: https://avatars3.githubusercontent.com/takakiyo
+author_github: https://github.com/takakiyo
+seo-title: Let's try Liberty InstantOn on Intel Mac - OpenLiberty.io
+seo-description: Try Liberty InstantOn on Intel Mac
```
```suggestion
seo-description: How developers can try Liberty InstantOn that enables millisecond startup of cloud-native Java applications on Intel Mac
```

yeekangc replied:
Cc @lauracowen
